### ISSUE TITLE
`--no-check` affects runtime behavior

### ISSUE THREAD

**Issue by wojpawlik** [denoland/deno#11437]
```ts
import XR from "https://cdn.skypack.dev/xregexp@5.0.2?dts";

const version = XR.tag("xn")`
  .+?
`;

console.log(version);
```

Without `--no-check`, correctly outputs `/.+?/`.
With `--no-check`, incorrectly outputs `/\n(?:).+?\n/`.

`deno 1.12.0+1ad6575 (canary, x86_64-pc-windows-msvc)`

Based on actual code: https://github.com/wojpawlik/deno2node/blob/d224750108fcdc70d666b3cc3f9e95e6310a4a47/src/_transformations/specifiers.ts#L5-L8.

Blocks #11340.

**Comment by kitsonk** [comment]
Upstream issue is: https://github.com/swc-project/swc/issues/1791

**Comment by bartlomieju** [comment]
Fixed in #11626
